### ISSUE TITLE
rest parameter not recognized in rules "no-undef" and "no-unused-vars"

### ISSUE THREAD

**Issue by niels4** [eslint/eslint#1870]
When validating these lines of javascript:
```javascript
var func = function (first, ...rest) {
  return rest
}

func(1, 2, 3)
```
with this .eslintrc config:

```javascript
{
    "rules": {
        "strict": false,
        "semi": 0
     },
    "ecmaFeatures": {
        "restParams": true,
        "spread": true
    }
}
```

I still get the following errors:
```
restParams.js
  1:21  error  first is defined but never used  no-unused-vars
  2:9   error  'rest' is not defined        no-undef

✖ 2 problems (2 errors, 0 warnings)
```

The rule doesn't seem to realize that there is a second parameter named 'rest' in the function declaration.

**Comment by niels4** [comment]
After digging around, looks like this is an issue on the side of espree not properly adding the rest parameter to the context.getScope().variables list. I'll move my issue to the es6 features discussion over there.

**Comment by nzakas** [comment]
This is an issue with escope, not Espree. See #1863 and #1543
